### PR TITLE
Task Description to Speech

### PR DESCRIPTION
**Pull request by mitcmurph**

### CODE REVIEW

File: unity/XRTProject/Assets/Scripts/Training/TrainingManager.cs [hunk 1]
```diff
@@ -187,6 +187,25 @@ public void SwitchTask(Task task)
...
+            {
+                if (obj is BackendErrorResponse error)
+                {
+                    Debug.LogError($"Error generating speech {error.Status}: {error.Message}");
+                }
+                else if (obj is byte[] bytes)
+                {
+                    AudioClip audioClip = WavUtility.ToAudioClip(bytes);
+                    AudioManager.instance.AddClip(audioClip, "taskDescription");
+                    AudioManager.instance.Play("taskDescription");
```
Comment (mitcmurph) [inline unity/XRTProject/Assets/Scripts/Training/TrainingManager.cs:202]: `AudioManager.instance...` works but should I be using `FindObjectOfType<AudioManager>()...`? Did not test with the latter

File: unity/XRTProject/Assets/Scripts/Sound/WavUtility.cs [hunk 2]
```diff
@@ -0,0 +1,417 @@
+﻿using UnityEngine;
```
Comment (mitcmurph) [inline unity/XRTProject/Assets/Scripts/Sound/WavUtility.cs:1]: External Library

File: unity/XRTProject/Assets/Scripts/Networking/ApiRequest.cs [hunk 3]
```diff
@@ -14,7 +14,7 @@ public class ApiRequest
     private string payload = string.Empty;
     private Action<UnityWebRequest> responseAction = null;
     private Action<BackendErrorResponse> errorAction = null;
-    private Action<string> responsePayload = null;
+    private Action<object> responsePayload = null;
```
Comment (mitcmurph) [inline unity/XRTProject/Assets/Scripts/Networking/ApiRequest.cs:17]: Prior to this change, all response payloads were returning `string`. Changed to `object` to allow for audio content as well.

File: unity/XRTProject/Assets/Scripts/Networking/ApiRequest.cs [hunk 4]
```diff
@@ -61,7 +61,7 @@ public ApiRequest HandleResponse(Action<object> responsePayload)
 
     public ApiRequest HandleJsonResponse<T>(Action<T> responsePayload)
     {
-        this.responsePayload = (p) => responsePayload.Invoke(JsonConvert.DeserializeObject<T>(p));
+        this.responsePayload = (p) => responsePayload.Invoke(JsonConvert.DeserializeObject<T>(p as string));
```
Comment (mitcmurph) [inline unity/XRTProject/Assets/Scripts/Networking/ApiRequest.cs:64]: cast `object` to `string`

**Review by kissmyuzi — APPROVED** [review]

**Pull request merged**